src/spyglass/position/v1/dlc_utils.py

```diff
@@ -714,30 +714,26 @@ def red_led_bisector_orientation(pos_df: pd.DataFrame, **params):
     LED2 = params.pop("led2", None)
     LED3 = params.pop("led3", None)
 
-    x_vec = pos_df[[LED1, LED2]].diff(axis=1).iloc[:, 0]
-    y_vec = pos_df[[LED1, LED2]].diff(axis=1).iloc[:, 1]
+    orient = np.full(len(pos_df), np.nan)  # Initialize with NaNs
+    x_vec = pos_df[LED1]["x"] - pos_df[LED2]["x"]
+    y_vec = pos_df[LED1]["y"] - pos_df[LED2]["y"]
 
-    y_is_zero = y_vec.eq(0)
-    perp_direction = pos_df[[LED3]].diff(axis=1)
+    # when y_vec is zero, 1&2 are equal. Compare to 3, determine if up or down
+    orient[y_vec.eq(0) & pos_df[LED3]["y"].gt(pos_df[LED1]["y"])] = np.pi / 2
+    orient[y_vec.eq(0) & pos_df[LED3]["y"].lt(pos_df[LED1]["y"])] = -np.pi / 2
 
-    # Handling the special case where y_vec is zero all Ys are the same
-    special_case = (
-        y_is_zero
-        & (pos_df[LED3]["y"] == pos_df[LED1]["y"])
-        & (pos_df[LED3]["y"] == pos_df[LED2]["y"])
-    )
-    if special_case.any():
+    # Handling error case where y_vec is zero and all Ys are the same
+    y_1, y_2, y_3 = pos_df[LED1]["y"], pos_df[LED2]["y"], pos_df[LED3]["y"]
+    if np.any(y_vec.eq(0) & y_1.eq(y_2) & y_1.eq(y_3)):
         raise Exception("Cannot determine head direction from bisector")
 
-    orientation = np.zeros(len(pos_df))
-    orientation[y_is_zero & perp_direction.iloc[:, 0].gt(0)] = np.pi / 2
-    orientation[y_is_zero & perp_direction.iloc[:, 0].lt(0)] = -np.pi / 2
-
-    orientation[~y_is_zero & ~x_vec.eq(0)] = np.arctan2(
-        y_vec[~y_is_zero], x_vec[~x_vec.eq(0)]
-    )
+    # General case where y_vec is not zero. Use arctan2 to determine orientation
+    length = np.sqrt(x_vec**2 + y_vec**2)
+    norm_x = (-y_vec / length)[~y_vec.eq(0)]
+    norm_y = (x_vec / length)[~y_vec.eq(0)]
```

Comment:
Just double checking that `norm_y` should use `x_vec`? I didn't look at the calculation too carefully.

Reply:
So, this `iterrows` versions has `x_vec` norm in the first arg of `arctan`, and `-y_vec` in the second arg
```python
norm = np.array([-y_vec / length, x_vec / length])
orientation.append(np.arctan2(norm[1], norm[0]))
```
Became this with she same order of args. By maybe my `norm` vars are mislabeled?
```
norm_x = (-y_vec / length)[~y_vec.eq(0)]
norm_y = (x_vec / length)[~y_vec.eq(0)]
orient[~y_vec.eq(0)] = np.arctan2(norm_y, norm_x)
```